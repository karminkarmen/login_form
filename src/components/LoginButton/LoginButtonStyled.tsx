import styled from 'styled-components';
import LoginButtonContainerStyled from './LoginButtonContainerStyled';
import { innerWidthSize } from '../../constants/themeConsts';

const LoginButtonStyled = styled.input.attrs(props => ({
    type: 'submit',
    value: 'login'
}))`
    width: ${innerWidthSize / 2}px;
    height: ${innerWidthSize / 2}px;
    border-radius: ${props => props.theme.radius.maximum};
    border: unset;
    text-transform: uppercase;
    letter-spacing: ${props => props.theme.letterSpacing.normal};
    background: transparent;
    color: ${props => props.theme.colors.secondary};
    font-size: ${props => props.theme.fontSizes.medium};
    font-family: ${props => props.theme.fontFamilies.lato};
    transition: ${props => props.theme.transitionTime.default};
    background: ${props => props.theme.colors.main};

    ${LoginButtonContainerStyled}:hover & {
        color: ${props => props.theme.colors.main};
        background-color: ${props => props.theme.colors.secondaryLighten};
        cursor: pointer;
    }
    &:focus {
        outline: none;
    }
`;

export default LoginButtonStyled;
