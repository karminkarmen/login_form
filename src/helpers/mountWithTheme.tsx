import React from 'react';
import { mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';
import theme from '../configs/theme';

const mountWithTheme = (child: React.ReactElement) => {
    return mount(child, {
        wrappingComponent: ({ children }) => <ThemeProvider theme={theme}>{children}</ThemeProvider>,
    });
}

export default mountWithTheme;