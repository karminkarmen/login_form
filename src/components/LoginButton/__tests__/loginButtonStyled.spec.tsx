import React from 'react';
import mountWithTheme from '../../../helpers/mountWithTheme';
import LoginButtonStyled from '../LoginButtonStyled';

describe('<LoginButtonStyled />', () => {

    it('matches snapshot', () => {
        const wrapper = mountWithTheme(<LoginButtonStyled />);
        expect(wrapper.debug()).toMatchSnapshot();
    });

    it('renders input with proper value and type', () => {
        const wrapper = mountWithTheme(<LoginButtonStyled />);

        expect(wrapper.find(`input[type='submit']`)).toHaveLength(1);
        expect(wrapper.find(`input[value='login']`)).toHaveLength(1);
    });

});
