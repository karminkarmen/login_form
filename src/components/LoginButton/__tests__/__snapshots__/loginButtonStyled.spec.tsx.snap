// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginButtonStyled /> matches snapshot 1`] = `"<input type=\\"submit\\" class=\\"sc-AxirZ ldlRsw\\" value=\\"login\\">"`;
